tests/runtest.py

```diff
@@ -845,6 +861,22 @@ def parse_argument():
     else:
         flags = arg.flags.split()
 
+    def has_compiler(compiler):
+        installed = os.system('command -v %s > /dev/null' % compiler) == 0
+        return installed
+
+    compilers = []
+    compilerslen = 0
+    if arg.compiler == 'all':
+        for compiler in ['gcc', 'clang']:
+            if has_compiler(compiler):
+                compilers.append(compiler)
+                compilerslen += 1
+    else:
+        if has_compiler(arg.compiler):
+            compilers.append(arg.compiler)
+            compilerslen += 1
```

Comment:
Do we really need to calculate the length separately?  Why not using `len(compilers)`?